### PR TITLE
Add accelerators workstation config resource

### PR DESCRIPTION
**Pull request by DanielRieske**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
workstations: added `accelerators` field to `google_workstations_workstation_config` resource(beta)
```

Closes: https://github.com/hashicorp/terraform-provider-google/issues/15311

### CODE REVIEW

**Comment by DanielRieske** [comment]
**Note:** I was not able to run all acceptances tests at the same time as my project had a quota of 5 on the amount of clusters that I could create within my region. I assume that the project that this is tested in has a higher quota otherwise I'll have to revist the tests.

`WorkstationClustersPerProjectPerRegion' exhausted. Limit 5 in region us-central1`

File: mmv1/products/workstations/WorkstationConfig.yaml [hunk 1]
```diff
@@ -263,6 +270,20 @@ properties:
                 description: |
                   Whether the instance has confidential compute enabled.
                 send_empty_value: true
+          - !ruby/object:Api::Type::Array
+            name: 'accelerators'
+            description: |
+              An accelerator card attached to the instance.
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
```
Comment (rileykarson) [inline mmv1/products/workstations/WorkstationConfig.yaml:278]: Can we make both subfields required? That'll avoid cases like the following:

```
accelerators {
  count = 1
}
```

File: mmv1/templates/terraform/examples/workstation_config_accelerators.tf.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  annotations = {
+    label-one = "value-one"
+  }
+}
+
+resource "google_workstations_workstation_config" "<%= ctx[:primary_resource_id] %>" {
+  provider               = google-beta
+  workstation_config_id  = "<%= ctx[:vars]['workstation_config_name'] %>"
+  workstation_cluster_id = google_workstations_workstation_cluster.<%= ctx[:primary_resource_id] %>.workstation_cluster_id
+  location   		         = "us-central1"
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/workstation_config_accelerators.tf.erb:35]: nit: spacing here is mixed tabs + spaces

Comment (DanielRieske) [reply mmv1/templates/terraform/examples/workstation_config_accelerators.tf.erb]: Not at all, I like to be consistent. Chaned it.

File: mmv1/products/workstations/WorkstationConfig.yaml (hunk 1 above)
Comment (DanielRieske) [reply mmv1/products/workstations/WorkstationConfig.yaml:278]: Made both of them required.

**Comment by DanielRieske** [comment]
Unsure why my changelog entry isn't going through the validator.

```
release-note:enhancement
workstations: added `accelerators` field to `google_workstations_workstation_config` resource(beta)

```

**Review by rileykarson — APPROVED** [review]

**Pull request merged**